linera-chain/src/manager.rs

```diff
@@ -163,19 +208,25 @@ impl MultiOwnerManager {
         }
         let expected_round = match validated {
             None => self.current_round(),
-            Some(cert) => cert.round.try_add_one()?.max(self.current_round()),
+            Some(cert) => self
+                .ownership
+                .next_round(cert.round)
+                .ok_or_else(|| ChainError::ArithmeticError(ArithmeticError::Overflow))?
+                .max(self.current_round()),
         };
         // In leader rotation mode, the round must equal the expected one exactly.
         // Only the first single-leader round can be entered at any time.
-        if new_round > self.multi_leader_rounds {
+        if self.is_super(owner)
+            || (new_round <= Round::SingleLeader(0) && !expected_round.is_fast())
+        {
             ensure!(
-                expected_round == new_round,
-                ChainError::WrongRound(expected_round)
+                expected_round <= new_round,
+                ChainError::InsufficientRound(expected_round)
             );
         } else {
             ensure!(
-                expected_round <= new_round,
-                ChainError::InsufficientRound(expected_round)
+                expected_round == new_round,
+                ChainError::WrongRound(expected_round)
             );
         }
         if let Some(old_proposal) = &self.proposed {
```

Comment:
typo below: `// We already accepted a proposal in this [round] or [at] a higher round.`